docs/digital/tutorials/circuits_with_bloqade.py

```diff
@@ -0,0 +1,524 @@
+#
+# The first is T state teleportation, which teleports a T gate that was applied to an ancilla (a "T state") onto the target state using only Clifford gates and feedforward. Due to the property of being Clifford, the circuit itself is fault tolerant and thus plays an important role in many error corrected algorithms.
+# %%
+@squin.kernel
+def t_teleport(target: squin.qubit.Qubit) -> squin.qubit.Qubit:
+    ancilla = squin.qubit.new(1)[0]
+    squin.gate.h(ancilla)
+    squin.gate.t(ancilla)
+    squin.gate.cx(control=target, target=ancilla)
+    bit = squin.qubit.measure(target)
+    if bit:
+        squin.gate.s(ancilla)
+    return ancilla  # The state of the target qubit is also teleported to the ancilla
+
+
+# And now lets wrap it into a larger context to run. In this case,
+# apply to a |+> state and see that we get a T|+> state out.
+@squin.kernel
+def t_teleport_wrapper() -> Register:
+
+    target = squin.qubit.new(1)[0]
+    squin.gate.h(target)
+    target = t_teleport(target)
+    return IList([target])
+
+
+# And run it
+emulator = StackMemorySimulator(min_qubits=2)
+task = emulator.task(t_teleport_wrapper)
+state = task.batch_state(shots=1000, qubit_map=lambda x: x)
+# Even though there is measurement and feedforward, the final state is still pure. Neat!
+print(state)
+# %% [markdown]
+# ### Constant depth GHZ state
+#
+# Remarkably, it is also possible to prepare a GHZ state with a constant number of gates. At first glance, this seems impossible: quantum information can only propagate as fast as the information lightcone, and so a constant depth circuit can only have a small amount of entanglement—not enough to prepare the long-range correlated GHZ state. The trick is, we can replace quantum gates with ancillas and classical feedforward measurements so that the information is propagated classically instead of quantumly. For more details, check out [1] and [2].
+#
+# [1] [Efficient Long-Range Entanglement Using Dynamic Circuits](https://doi.org/10.1103/PRXQuantum.5.030339)
+#
+# [2] [Constant-Depth Preparation of Matrix Product States with Adaptive Quantum Circuits](https://doi.org/10.1103/PRXQuantum.5.030344)
+#
+# The explicit circuit for the GHZ circuit is shown in Fig. 5 of [1]. There is classical feedforward in the form of a parity check, which requires a classical XOR operation that is irrepresentable by CIRQ.
+
+
+# %%
+def ghz_constant_depth(n_qubits: int):
+
+    @squin.kernel
+    def main() -> Register:
+        qreg = squin.qubit.new(n_qubits)
+        ancilla = squin.qubit.new(n_qubits - 1)
+        h = squin.op.h()
+        cx = squin.op.cx()
+        x = squin.op.x()
+
+        for i in range(n_qubits):
+            squin.qubit.apply(h, qreg[i])
+
+        for i in range(n_qubits - 1):
+            squin.qubit.apply(cx, [qreg[i], ancilla[i]])
```

Comment:
Update the signature here (and below) to `squin.qubit.apply(cx, qreg[i], ancilla[i])`.